src/java.base/share/classes/java/nio/X-Buffer.java.template

```diff
@@ -2042,6 +2044,13 @@ public abstract sealed class $Type$Buffer
      * @since  1.5
      */
     public $Type$Buffer append(CharSequence csq, int start, int end) {
+        if (csq instanceof CharBuffer cb) {
```

Comment:
This change means the code no longer checks there is sufficient space so IOBE is thrown instead of BufferedOverException when there is sufficient space. It's a reminder that we'll need to expand the tests in advance of, or as part of, any of optimisations in this area.